docs/tutorials/tpu-cluster-setup.md

```diff
@@ -0,0 +1,150 @@
+# Setting up a TPU Cluster
+
+This guide will walk you through the steps to set up and manage a TPU cluster using [Google Cloud Platform (GCP)](https://cloud.google.com/tpu/docs/ray-guide) and [Ray](https://docs.ray.io/en/latest/cluster/vms/user-guides/launching-clusters/gcp.html). Currently, this is the only way to use multinode TPU (i.e. more than a v4-8) with Marin.
+## Prerequisites
+
+Before you begin, ensure you have:
+- Completed the [basic installation](installation.md)
+- Access to a Google Cloud project with TPU quota
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) installed
+- [Ray](https://docs.ray.io/en/latest/installation.html) installed
+
+## TPU Cluster Architecture
+
+A typical TPU cluster consists of:
+- **Head Node**: A persistent GCP VM that coordinates the cluster
+- **Worker Nodes**: Autoscaling TPU VMs that execute the actual computation
+
+## Cluster Configuration
+
+The cluster configuration is defined in a [YAML template file](../../infra/marin-cluster-template.yaml). This template allows you to customize various aspects of your cluster:
```

Comment:
```suggestion

Ray's autoscaler is configured using YAML. Please refer to [Ray's full documentation on autscaling on GCP](https://docs.ray.io/en/latest/cluster/vms/user-guides/launching-clusters/gcp.html) for more details. We provide a starting point that should work in many common cases. 


!!! tip
     In particular, we have not tested [KubeRay](https://github.com/ray-project/kuberay) because it does not work with [TRC](https://sites.research.google/trc/about/)-provided TPUs.

The cluster configuration is defined in a [YAML template file](../../infra/marin-cluster-template.yaml). This template allows you to customize various aspects of your cluster:
```